platforms/android/build_sdk.py

```diff
@@ -48,6 +48,53 @@ def check_dir(d, create=False, clean=False):
             os.makedirs(d)
     return d
 
+def which(cmd, mode=os.F_OK | os.X_OK, path=None):
+
+    def _access_check(fn, mode):
+        return os.path.exists(fn) and os.access(fn, mode) and not os.path.isdir(fn)
+
+    # If we're given a path with a directory part, look it up directly
+    # rather than referring to PATH directories. This includes checking
+    # relative to the current directory, e.g. ./script
+    if os.path.dirname(cmd):
+        if _access_check(cmd, mode):
+            return cmd
+
+        return None
+
+    if path is None:
+        path = os.environ.get("PATH", os.defpath)
+    if not path:
+        return None
+
+    path = path.split(os.pathsep)
+
+    if sys.platform == "win32":
+        # The current directory takes precedence on Windows.
+        if os.curdir not in path:
+            path.insert(0, os.curdir)
+
+            # need to consider PATHEXT on Windows.
+            pathext = os.environ.get("PATHEXT", "").split(os.pathsep)
+            if any(cmd.lower().endswith(ext.lower()) for ext in pathext):
+                files = [cmd]
+            else:
+                files = [cmd + ext for ext in pathext]
+    else:
```

Comment:
Code is not complete.

I believe we can use something like this instead:
```
def check_executable(cmd)
    try:
        retcode = subprocess.call(cmd, shell=shell)
        if retcode < 0:
            return False
        return True
    except:
        return False

if not check_executable(['ninja', "--version"])
    ... fallback for manual searching ...
if not check_executable(['cmake', "--version"])
    ...
```

Reply:
Will add something like this and push a new commit

Reply:
Pushed a new commit to address this.
I tested the changes on Windows and Mac and they seem to work well